uiServices/src/testSuiteServices.go

```diff
@@ -84,14 +89,169 @@ func ValidateJsonWithSchema(testSuite []byte, schemaName, structType string) boo
+
+	if err != nil {
+		logrus.Error("Cannot Unmarshall Json")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if !testSuiteWriterXml(testSuite, testSuitePathDir) {
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	rw.WriteHeader(http.StatusNoContent)
+}
+
+func getTestSuite(rw http.ResponseWriter, req *http.Request){
+
+	testSuitePathDir := getTestSuiteHeader(req)
+	testSuiteName := chi.URLParam(req, "testSuiteName")
+
+	ValidateFileNameAndHeader(rw,req,testSuitePathDir, testSuiteName)
+
+	file, err := os.Open(fmt.Sprintf("%s%s.xml", testSuitePathDir, testSuiteName))
+	if err != nil {
+		logrus.Error("Test Suite Name Not Found: "+testSuitePathDir + testSuiteName)
+		rw.WriteHeader(http.StatusNotFound)
+		return
+	}
+	defer file.Close()
+
+
+	var testSuite testStrategies.TestSuite
+
+	byteValue, err := ioutil.ReadAll(file)
+	if err != nil{
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Read File")
+		return
+	}
+
+	err = xml.Unmarshal(byteValue, &testSuite)
+	if err != nil{
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Unmarshall")
+		return
+	}
+
+	testSuiteJSON, err := json.MarshalIndent(testSuite,"","")
+	if err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Marshall")
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+	rw.Write(testSuiteJSON)
+	logrus.Println(string(testSuiteJSON))
+
+}
+
+func getAllTestSuites(rw http.ResponseWriter, req *http.Request){
+	var testSuite testStrategies.TestSuite
+	var filename string
+
+	type Suite struct {
+		File string `json:"file"`
+		Name string `json:"name"`
+		Description string `json:"description"`
+	}
+
+	var suite Suite
+
+	type Suites[]Suite
+
+	suites := Suites{}
+
+	testSuitePathDir := getTestSuiteHeader(req)
+	if len(testSuitePathDir) <= 1 {
+		logrus.Error("No file directory entered")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	files, err := ioutil.ReadDir(testSuitePathDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, file := range files {
+		if filepath.Ext(testSuitePathDir + file.Name()) == ".xml" {
+			filename = file.Name()
+
+			file, err := os.Open(fmt.Sprintf("%s%s", testSuitePathDir, filename))
+			if err != nil {
+				logrus.Error("Cannot open file: ", filename)
+			}
+
+			byteValue, err := ioutil.ReadAll(file)
+			if err != nil{
+				logrus.Error("Cannot Read File: ", filename)
+			}
+
+			err = xml.Unmarshal(byteValue, &testSuite)
+			if err != nil {
+				logrus.Error("Cannot Unmarshall: ", filename)
+			}
+
+			suite.File = filename
+			suite.Name = testSuite.Name
+			suite.Description = testSuite.Description
+
+			//if a Test Suite Name can't be assigned, it isn't a Test Suite object
+			if suite.Name != "" {
+				suites = append(suites, suite)
+			}
+
+			//ensure values are reset every iteration
+			filename = ""
+			testSuite.Name = ""
+			testSuite.Description = ""
+
+		}
+	}
+
+	json.NewEncoder(rw).Encode(suites)
```

Comment:
check for errors

Reply:
This line will encode Test Suites which will had to have been validated to be added to "suites" so there should be no error. If "suites" is empty, then a blank array is returned, which I think is ok.  

What do you mean check for errors here? Or is it this line you mean to check?

Reply:
https://golang.org/pkg/encoding/json/#Encoder.Encode returns an error. we should check that it has encoded succesfully before we return a 200

Reply:
Something like this:

	err = json.NewEncoder(rw).Encode(suites)
	if err != nil {
		logrus.Error("Could Not Encode Test Suites")
		rw.WriteHeader(http.StatusInternalServerError)
		return
	}

	rw.WriteHeader(http.StatusOK)